src/lib/FieldType/Validator/TargetContentValidatorInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Core\FieldType\Validator;
+
+use eZ\Publish\Core\FieldType\ValidationError;
+
+interface TargetContentValidatorInterface
```

Comment:
I think it would a good candidate for something more generic, something that could evolve into proper FieldType Validator in the future. Then you could pass Value and Validator configuration array and let the Validator to extract Content ID and allowed CTs. We wouldn't be limited to just a single use case here but we could reuse this Interface for other validators in the future.

Reply:
Do we want to do this in the following PR?

Reply:
Do you think you can at least make the interface fieldtype agnostic and keep implementation for Relation FT for now?

cc @alongosz @adamwojs @Steveb-p

Reply:
@webhdx @adamwojs resolved in https://github.com/ezsystems/ezpublish-kernel/pull/3131/commits/0cb01ba75318f9222df1331f04f94e89c9a3ff38